### PR TITLE
Fixes #28363 - Drop lookup key matcher length limit

### PR DESCRIPTION
**Pull request by shiramax**

### CODE REVIEW

**Review by tbrisker — CHANGES_REQUESTED** [review]
test failures related

File: test/models/shared/taxonomies_base_test.rb [hunk 1]
```diff
@@ -392,29 +392,13 @@ module TaxonomiesBaseTest
...
     end
-
-    test "taxonomy should not save when matcher is exactly 256 characters" do
-      parent = FactoryBot.create(:"#{taxonomy_name}", :name => 'a' * (255 - taxonomy_name.length - 2))
-      taxonomy = taxonomy_class.new :parent => parent, :name => 'b'
-      refute_valid taxonomy
-      assert_equal _("is too long (maximum is 0 characters)"), taxonomy.errors[:name].first
-    end
-
+
```
[the remark this answers is not in the document]
Comment (tbrisker) [reply test/models/shared/taxonomies_base_test.rb]: This is blocking tests from running

**Comment by tbrisker** [comment]
test failure related

**Review by tbrisker — APPROVED** [review]
Thanks @shiramax !

**Comment by shiramax** [comment]
[test katello]

**Pull request merged**